### PR TITLE
drone: 1.6.5 → 1.9.0

### PR DESCRIPTION
**Pull request by blaggacao**

### CODE REVIEW

**Comment by blaggacao** [comment]
Result of `nixpkgs-review pr 93681` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- drone
</details>

**Comment by blaggacao** [comment]
```console
[nix-shell:~/.cache/nixpkgs-review/pr-93681-1]$ ./results/drone/bin/drone-
drone-agent       drone-controller  drone-server

[nix-shell:~/.cache/nixpkgs-review/pr-93681-1]$ ./results/drone/bin/drone-agent --help
Usage of ./results/drone/bin/drone-agent:
  -env-file string
    	Read in a file of environment variables (default ".env")

[nix-shell:~/.cache/nixpkgs-review/pr-93681-1]$ ./results/drone/bin/drone-controller --help
{"error":"strconv.ParseInt: parsing \"\": invalid syntax","level":"fatal","msg":"cannot parse stage ID","time":"2020-07-22T17:38:05-05:00"}

[nix-shell:~/.cache/nixpkgs-review/pr-93681-1]$ ./results/drone/bin/drone-server --help
Usage of ./results/drone/bin/drone-server:
  -env-file string
    	Read in a file of environment variables (default ".env")
```

**Comment by blaggacao** [comment]
So `vendorSha256` didn't change.

And `fetchGithub.sha256` apparently is ignored anyway. I don't think so.
But it works.

File: pkgs/development/tools/continuous-integration/drone/default.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 buildGoModule rec {
   name = "drone.io-${version}";
-  version = "1.6.5";
+  version = "1.9.0";
```
Comment (elohmeier) [inline pkgs/development/tools/continuous-integration/drone/default.nix:5]: The sha256 in the src attribute and probably the vendorSha256 need to be updated as well.

**Review by elohmeier — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/continuous-integration/drone/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/development/tools/continuous-integration/drone/default.nix:5]: I'm suprised too: Neither of them. I faked both and they resulted the same.

Comment (elohmeier) [reply pkgs/development/tools/continuous-integration/drone/default.nix:5]: It should be 
```diff
-  vendorSha256 = "1dvf8vz3jr9smki3jql0kvd8z8rwdq93y7blbr2yjjfsdvx6lxl1";
+  vendorSha256 = "0idf11sr417lxcjryplgb87affr6lgzxazzlyvk0y40hp8zbhwsx";

   src = fetchFromGitHub {
     owner = "drone";
     repo = "drone";
     rev = "v${version}";
-    sha256 = "05cgd72qyss836fby0adhrm5p8g7639psk2yslhg6pmz0cqfbq9m";
+    sha256 = "1lsyd245fr1f74rpccvvw41h5g75b79afrb8g589bj13ggjav0xy";
   };
```

Comment (blaggacao) [reply pkgs/development/tools/continuous-integration/drone/default.nix:5]: Done. That's still _darknix_ to me. Why https://github.com/NixOS/nixpkgs/pull/93681#issuecomment-662732113, then?

**Review by elohmeier — APPROVED** [review]

File: pkgs/development/tools/continuous-integration/drone/default.nix (hunk 1 above)
Comment (elohmeier) [reply pkgs/development/tools/continuous-integration/drone/default.nix:5]: Nix picks up existing (cached) versions of the `src` and the `vendor` go module dependencies based on the hash from the nix store. It won't notice if e.g. that cached source contains a different version of drone than intended.

**Review by danieldk — APPROVED** [review]
LGTM

Result of `nixpkgs-review pr 93681` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- drone
</details>

Binaries run as expected.

**Pull request merged**